modules/javafx.graphics/src/main/java/com/sun/glass/ui/win/WinTextRangeProvider.java

```diff
@@ -98,8 +100,28 @@ private void validateRange(String text) {
         }
 
         int length = text.length();
-        start = Math.max(0, Math.min(start, length));
-        end = Math.max(start, Math.min(end, length));
+        start = Utils.clamp(0, start, length);
+        end = Utils.clamp(start, end, length);
```

Comment:
This is only cleanup and not required for this fix.

Reply:
I applied the software engineering principle to leave the code cleaner than seen. (Martin Fowler et all) 

Should I revert this?

Reply:
Sorry for not being clear. I just wanted to point out that it is not required for the issue fix.
Please don't revert. This looks better.

Reply:
This is the sort of tangentially-related change that should usually be avoided. In this specific case, `clamp(min, val, max)` will produce the same results as `max(start, min(val, end))`, but it isn't guaranteed to do so for malformed input where `max < min`. Given that `text.length()` always returns a non-negative number, the first call to `clamp` is identical to the existing code. And since the resulting` start` value is guaranteed to be `<= length`, the second call to `clamp` is also equivalent.

But... reviewers shouldn't have to spend time verifying this to ensure no regression. Since Ambarish is the primary reviewer of this, and he is OK with this change (and I already did spend the time to verify that it is equivalent), then I don't insist that you revert it.